pkg/lib/oauth/handler/handler_token.go

```diff
@@ -364,6 +364,13 @@ func (h *TokenHandler) validateRequest(r protocol.TokenRequest, client *config.O
 		// The validation logics can be different depends on requested_token_type
 		// Do the validation in methods for each requested_token_type
 		break
+	case oauth.ClientCredentialsGrantType:
+		if r.Resource() == "" {
+			return protocol.NewError("invalid_request", "resource is required")
+		}
+		if r.ClientSecret() == "" {
+			return protocol.NewError("invalid_request", "client secret is required")
```

Comment:
According to https://datatracker.ietf.org/doc/html/rfc6749#section-5.2 when client secret is required but not provided, `invalid_client` `client secret is required` should be returned.

Reply:
It should be OK, since this error indicates a programming error anyway, and it is not a recoverable error in a running deployment.

Reply:
Added a commit to change the error to `invalid_client`